src/include/daos_srv/rsvc.h

```diff
@@ -135,29 +135,40 @@ enum ds_rsvc_start_mode {
 	DS_RSVC_DICTATE	/**< DANGEROUSLY reset and start the service (see rdb_dictate) */
 };
 
+/* clang-format off */
 int ds_rsvc_start(enum ds_rsvc_class_id class, d_iov_t *id, uuid_t db_uuid, uint64_t caller_term,
-		  enum ds_rsvc_start_mode mode, size_t size, uint32_t vos_df_version,
-		  d_rank_list_t *replicas, void *arg);
+		  enum ds_rsvc_start_mode mode, struct rdb_create_params *create_params, void *arg);
 int ds_rsvc_stop(enum ds_rsvc_class_id class, d_iov_t *id, uint64_t caller_term, bool destroy);
 int ds_rsvc_stop_leader(enum ds_rsvc_class_id class, d_iov_t *id,
 			struct rsvc_hint *hint);
+/* clang-format on */
+
+/** Parameters used for creating an rsvc */
+struct ds_rsvc_create_params {
```

Comment:
no change requested, though it is so tempting to me to consider if this structure definition and struct rdb_create_params can be consolidated into one common structure. On "this side" (initiating) of the bcast RPC maybe self id may not make perfect sense, and on the handling side bootstrap may not make perfect sense.

Reply:
Yeah, I used a common struct at the beginning, and then confused myself because of the differences you pointed out sharply. Moreover, the replica IDs are required regardless of `bootstrap` when creating an rsvc, but are not required when creating an rdb replica without bootstrapping it---it might be awkward to move code that tries to find out the self ID from an array of replicas IDs from `rsvc` to `rdb`. Hope I won't regret about this later. :)